Solaris French locales classify some high-byte characters differently from
the upstream PCRE2 locale-test outputs. RunTest already expects that there
could be multiple correct answers which are stored in testoutput3*. Add a
Solaris-specific alternate without changing the existing upstream outputs.

Upstream pull request: https://github.com/PCRE2Project/pcre2/pull/973

Solaris fr treats these bytes as alphabetic/word characters:

  \xb2  ²  superscript two
  \xb3  ³  superscript three
  \xb9  ¹  superscript one
  \xbc  ¼  one quarter
  \xbd  ½  one half
  \xbe  ¾  three quarters

--- pcre2-10.48.orig/RunTest	2026-08-31 15:49:01.000000000 +0200
+++ pcre2-10.48/RunTest	2026-09-07 16:39:51.827532650 +0200
@@ -620,6 +620,7 @@
             outfile2=$testdata/testoutput3A
             outfile3=$testdata/testoutput3B
             outfile4=$testdata/testoutput3C
+            outfile5=test3outputD
           else
             infile=test3input
             outfile=test3output
@@ -626,6 +627,7 @@
             outfile2=test3outputA
             outfile3=test3outputB
             outfile4=test3outputC
+            outfile5=test3outputD
             sed "s/fr_FR/$loc/" $testdata/testinput3 >test3input
             sed "s/fr_FR/$loc/" $testdata/testoutput3 >test3output
             sed "s/fr_FR/$loc/" $testdata/testoutput3A >test3outputA
@@ -632,6 +634,14 @@
             sed "s/fr_FR/$loc/" $testdata/testoutput3B >test3outputB
             sed "s/fr_FR/$loc/" $testdata/testoutput3C >test3outputC
           fi
+          sed "s/fr_FR/$loc/" $testdata/testoutput3D >test3outputD.utf8
+          charmap=`LC_ALL=$loc locale charmap 2>/dev/null`
+          if command -v iconv >/dev/null 2>&1 && [ "$charmap" != "" ] && [ "$charmap" != "UTF-8" ]; then
+            iconv -f UTF-8 -t "$charmap" test3outputD.utf8 >test3outputD || cp test3outputD.utf8 test3outputD
+          else
+            cp test3outputD.utf8 test3outputD
+          fi
+          rm -f test3outputD.utf8
           break
         fi
       fi
@@ -649,7 +659,8 @@
           if $cf $outfile testoutput$bits/testoutput3 >teststdout || \
              $cf $outfile2 testoutput$bits/testoutput3 >>teststdout || \
              $cf $outfile3 testoutput$bits/testoutput3 >>teststdout || \
-             $cf $outfile4 testoutput$bits/testoutput3 >>teststdout
+             $cf $outfile4 testoutput$bits/testoutput3 >>teststdout || \
+             $cf $outfile5 testoutput$bits/testoutput3 >>teststdout
           then
             echo "  OK$with"
           else

--- /dev/null	2026-09-07 16:39:00.000000000 +0200
+++ pcre2-10.48/testdata/testoutput3D	2026-09-07 16:39:51.823094922 +0200
@@ -0,0 +1,183 @@
+# This set of tests checks local-specific features, using the "fr_FR" locale. 
+# It is almost Perl-compatible. When run via RunTest, the locale is edited to
+# be whichever of "fr_FR", "french", or "fr" is found to exist. There is
+# different version of this file called wintestinput3 for use on Windows,
+# where the locale is called "french" and the tests are run using
+# RunTest.bat. 
+
+#forbid_utf
+
+/^[\w]+/
+\= Expect no match
+    École
+No match
+
+/^[\w]+/locale=fr_FR
+    École
+ 0: \xc9cole
+
+/^[\W]+/
+    École
+ 0: \xc9
+
+/^[\W]+/locale=fr_FR
+\= Expect no match
+    École
+No match
+
+/[\b]/
+    \b
+ 0: \x08
+\= Expect no match
+    a
+No match
+
+/[\b]/locale=fr_FR
+    \b
+ 0: \x08
+\= Expect no match
+    a
+No match
+
+/^\w+/
+\= Expect no match
+    École
+No match
+
+/^\w+/locale=fr_FR
+    École
+ 0: \xc9cole
+
+/(.+)\b(.+)/
+    École
+ 0: \xc9cole
+ 1: \xc9
+ 2: cole
+
+/(.+)\b(.+)/locale=fr_FR
+\= Expect no match
+    École
+No match
+
+/École/i
+    École
+ 0: \xc9cole
+\= Expect no match
+    école
+No match
+
+/École/i,locale=fr_FR
+    École
+ 0: \xc9cole
+    école
+ 0: \xe9cole
+
+/\w/I
+Capture group count = 0
+Starting code units: 0 1 2 3 4 5 6 7 8 9 A B C D E F G H I J K L M N O P
+  Q R S T U V W X Y Z _ a b c d e f g h i j k l m n o p q r s t u v w x y z
+Subject length lower bound = 1
+
+/\w/I,locale=fr_FR
+Capture group count = 0
+Starting code units: 0 1 2 3 4 5 6 7 8 9 A B C D E F G H I J K L M N O P
+  Q R S T U V W X Y Z _ a b c d e f g h i j k l m n o p q r s t u v w x y z
+  \xaa \xb2 \xb3 \xb5 \xb9 \xba \xbc \xbd \xbe \xc0 \xc1 \xc2 \xc3 \xc4 \xc5
+  \xc6 \xc7 \xc8 \xc9 \xca \xcb \xcc \xcd \xce \xcf \xd0 \xd1 \xd2 \xd3 \xd4
+  \xd5 \xd6 \xd8 \xd9 \xda \xdb \xdc \xdd \xde \xdf \xe0 \xe1 \xe2 \xe3 \xe4
+  \xe5 \xe6 \xe7 \xe8 \xe9 \xea \xeb \xec \xed \xee \xef \xf0 \xf1 \xf2 \xf3
+  \xf4 \xf5 \xf6 \xf8 \xf9 \xfa \xfb \xfc \xfd \xfe \xff
+Subject length lower bound = 1
+
+# All remaining tests are in the fr locale, so set the default.
+
+#pattern locale=fr_FR
+
+/^[\xc8-\xc9]/i
+    École
+ 0: \xc9
+    école
+ 0: \xe9
+
+/^[\xc8-\xc9]/
+    École
+ 0: \xc9
+\= Expect no match
+    école
+No match
+
+/\xb5/i
+    µ
+ 0: \xb5
+\= Expect no match
+    \x9c
+No match
+
+/ÿ/i
+    \xff
+ 0: \xff
+\= Expect no match
+    y
+No match
+
+/(.)\1/i
+    \xfe\xde
+ 0: \xfe\xde
+ 1: \xfe
+
+/\W+/
+    >>>\xaa<<<
+ 0: >>>
+    >>>\xba<<< 
+ 0: >>>
+
+/[\W]+/
+    >>>\xaa<<<
+ 0: >>>
+    >>>\xba<<< 
+ 0: >>>
+
+/[^[:alpha:]]+/
+    >>>\xaa<<<
+ 0: >>>
+    >>>\xba<<< 
+ 0: >>>
+
+/\w+/
+    >>>\xaa<<<
+ 0: \xaa
+    >>>\xba<<< 
+ 0: \xba
+
+/[\w]+/
+    >>>\xaa<<<
+ 0: \xaa
+    >>>\xba<<< 
+ 0: \xba
+
+/[[:alpha:]]+/
+    >>>\xaa<<<
+ 0: \xaa
+    >>>\xba<<< 
+ 0: \xba
+    
+/[[:alpha:]][[:lower:]][[:upper:]]/IB
+------------------------------------------------------------------
+        Bra
+        [A-Za-z\xaa\xb2\xb3\xb5\xb9\xba\xbc-\xbe\xc0-\xd6\xd8-\xf6\xf8-\xff]
+        [a-z\xb5\xdf-\xf6\xf8-\xff]
+        [A-Z\xc0-\xd6\xd8-\xde]
+        Ket
+        End
+------------------------------------------------------------------
+Capture group count = 0
+Starting code units: A B C D E F G H I J K L M N O P Q R S T U V W X Y Z
+  a b c d e f g h i j k l m n o p q r s t u v w x y z \xaa \xb2 \xb3 \xb5 \xb9
+  \xba \xbc \xbd \xbe \xc0 \xc1 \xc2 \xc3 \xc4 \xc5 \xc6 \xc7 \xc8 \xc9 \xca
+  \xcb \xcc \xcd \xce \xcf \xd0 \xd1 \xd2 \xd3 \xd4 \xd5 \xd6 \xd8 \xd9 \xda
+  \xdb \xdc \xdd \xde \xdf \xe0 \xe1 \xe2 \xe3 \xe4 \xe5 \xe6 \xe7 \xe8 \xe9
+  \xea \xeb \xec \xed \xee \xef \xf0 \xf1 \xf2 \xf3 \xf4 \xf5 \xf6 \xf8 \xf9
+  \xfa \xfb \xfc \xfd \xfe \xff
+Subject length lower bound = 3
+
+# End of testinput3 
